src/conv/invokers/impl_gemm_dynamic.cpp

```diff
@@ -107,6 +229,39 @@ InvokerFactory MakeImplGemmDynamicDataInvokerFactory(const ConvolutionContext& c
             };
         };
     }
+    else if(ctx.direction.IsBackwardData())
+    {
+        return [ctx](const std::vector<Kernel>& kernels) {
+            return [=](const Handle& handle, const boost::any& primitive_parameters) {
+                const auto data_ctx = boost::any_cast<conv::DataInvokeParams>(primitive_parameters);
+                const auto& tensors = data_ctx.tensors;
+                auto kernel         = handle.Run(kernels[0]);
+                if(kernel.GetName().find("igemm_bwd_gtc") == 0)
+                {
+                    std::vector<KernelInvoke> ks;
+                    std::transform(kernels.begin(),
+                                   kernels.end(),
+                                   std::back_inserter(ks),
+                                   [&](const Kernel& k) { return handle.Run(k); });
+                    float elapsed = 0;
+
+                    elapsed = CallImplicitGemmDynamic(
```

Comment:
I would suggest updating CallImplicitGemmDynamic to take invoke params in some form instead of ConvolutionContext. ConvolutionContext may contain some outdated data from find time, depending on what you would use inside. Someone may accidentally use some data from the context which may be outdated and may not be. That would lead to problems that may be hard to dissect.

Reply:
Hi @DrizztDoUrden thanks for pointing out. I've think about this, but when doing CallImplicitGemmDynamic, I only need to query from ConvolutionContext is the tensor size (`h,w,n,c,k,filter_y,filter_x,stride_y...`). So if need to replace ConvolutionContext with some kernel arg, I can think of create a data structure like below (ImplicitGemmDynamicInvokeArgCommon), and filling the member from the member of ConvolutionContext. This way, I think this structure is a little bit redundant.
Do you still think I need create this structure? Or if I'm not understand correctly.
```
struct ImplicitGemmDynamicInvokeArgCommon{
    int hi;
    int wi;
    int n;
    int k;
    int c;
    int ho;
    int wo;
    int stride_h;
    int stride_w;
    int dilation_h;
    int dilation_w;
    int pad_h;
    int pad_w;
    int y;
    int x;
};
```

Reply:
I think you could pass there ConvDataTensors, DataInvokeParams or boost::any containing the DataInvokeParams/WrWInvokeParams (which would soon be replaced with AnyInvokeParams), depending on whether both wrw and data are supported at the same time, and fetch the data from them rather than conv context inside.
Another option is to fetch conv::ProblemDescription from ConvolutionContext, which should not contain any outdated info at any point.

Reply:
@DrizztDoUrden OK let me think about writing code in this way.

Reply:
@DrizztDoUrden Hi I choose to fetch conv::ProblemDescription from ConvolutionContext, and re-factor the invoker functions. Please help check if that is desired.